### PR TITLE
Removes cluster node status reporting from agent

### PR DESCRIPTION
**Pull request by 0xavi0**
Removes cluster node status reporting from agent and adds a basic unit test for the `clusterstatus Ticker`

Refers to: https://github.com/rancher/fleet/issues/1973

### CODE REVIEW

File: internal/cmd/agent/clusterstatus/ticker_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+		ctrl := gomock.NewController(GinkgoT())
+		clusterClient = fake.NewMockClientInterface[*fleet.Cluster, *fleet.ClusterList](ctrl)
+		nodeClient = fake.NewMockNonNamespacedClientInterface[*v1.Node, *v1.NodeList](ctrl)
+		agentNamespace = "cattle-fleet-system"
+		clusterName = "cluster-name"
+		clusterNamespace = "cluster-namespace"
+		baseTime = metav1.Now()
+		ctx, cancel = context.WithCancel(context.TODO())
+		clusterClient.EXPECT().Patch(clusterNamespace, clusterName, types.MergePatchType, gomock.Any(), "status").
+			DoAndReturn(func(namespace, name string, pt types.PatchType, data []byte, subresources ...string) (result fleet.Cluster, err error) {
```
[the remark this answers is not in the document]
Comment (0xavi0) [reply internal/cmd/agent/clusterstatus/ticker_test.go]: indeed, changed.

File: internal/cmd/agent/clusterstatus/ticker_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+				}
+				// only storing the lastseen time value here.
+				// We're not checking for values here because calling Expect,
+				// for example, makes the mock call panic when it doesn't succeed
+				lastTime = cluster.Status.Agent.LastSeen
+				agentStatusNamespace = cluster.Status.Agent.Namespace
+				return *cluster, nil
+			}).AnyTimes()
+		Ticker(ctx, agentNamespace, clusterNamespace, clusterName, time.Second*1, nodeClient, clusterClient)
+	})
```
Comment (manno) [inline internal/cmd/agent/clusterstatus/ticker_test.go:55]: nit: I'd always put newlines in between the blocks, but 🤷 
```suggestion
	})
	
```

Comment (0xavi0) [reply internal/cmd/agent/clusterstatus/ticker_test.go:56]: Sure, it's more readable that way. New lines added.

File: internal/cmd/agent/clusterstatus/ticker.go [hunk 3]
```diff
@@ -72,30 +70,11 @@ func Ticker(ctx context.Context,
 
 // Update the cluster.fleet.cattle.io status in the upstream cluster with the current node status
 func (h *handler) Update() error {
-	nodes, err := h.nodes.List(metav1.ListOptions{})
```
[the remark this answers is not in the document]
Comment (0xavi0) [reply internal/cmd/agent/clusterstatus/ticker.go:75]: Addressed as suggested

File: internal/cmd/agent/clusterstatus/ticker_test.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+	It("Increases the timestamp used to call Patch", func() {
+		By("Comparing every 2 seconds for a 6 seconds period")
+		Consistently(func() bool {
+			// return true when we're calling before Path was even called
```
[the remark this answers is not in the document]
Comment (0xavi0) [reply internal/cmd/agent/clusterstatus/ticker_test.go]: oops! Should be fixed now.

File: internal/cmd/agent/clusterstatus/ticker.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (weyfonk) [reply internal/cmd/agent/clusterstatus/ticker.go:75]: I meant _log messages_ , eg [here](https://github.com/rancher/fleet/pull/2190/files#diff-b4df31de0b406484919c188529ea36ed8494f210e7348ecaf0b522db37e29cdcR53) and in another goroutine in `Ticker()` ;)

[the remark this answers is not in the document]
Comment (weyfonk) [reply internal/cmd/agent/clusterstatus/ticker.go:75]: There's a couple left [here](https://github.com/rancher/fleet/pull/2190/files#diff-b4df31de0b406484919c188529ea36ed8494f210e7348ecaf0b522db37e29cdcR63) and 2 lines below.

[the remark this answers is not in the document]
Comment (0xavi0) [reply internal/cmd/agent/clusterstatus/ticker.go:75]: Hopefully fixed now

[the remark this answers is not in the document]
Comment (weyfonk) [reply internal/cmd/agent/clusterstatus/ticker.go:75]: Thanks :relaxed:

**Review by weyfonk — APPROVED** [review]

**Pull request merged**